puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -38,7 +38,7 @@ def report_to_hash
       end
 
       resources = build_resources_list
-      is_noop = resources.any? { |rs| has_noop_event?(rs) } and resources.none? { |rs| has_failed_event?(rs) }
+      is_noop = resources.any? { |rs| has_noop_event?(rs) } && resources.none? { |rs| has_enforcement_event?(rs) }
```

Comment:
Don't we now have the possibility to just simplify this, and scan all resource events to make sure they are noop? This would make the logic a little faster as well I imagine.